kratos/includes/model_part.h

```diff
@@ -1574,6 +1633,40 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief This method trims a string and returns the first component
+     * @param rStringName The given name to be trimmed
+     * @return The first index
+     */
+    IndexType FirstComponentName(const std::string& rStringName)
```

Comment:
I don't think you need the non-const version

Reply:
OK